### PR TITLE
[ShadowLayer] Mark shadowPath invalid on cornerRadius change

### PR DESCRIPTION
**Pull request by ianegordon**
Note the cornerRadius changes are not getting reflected in the before version.

| Before | After |
|--|--|
|shadowradius-broken|shadowradius-fixed|

### CODE REVIEW

File: components/ShadowLayer/src/MDCShadowLayer.m [hunk 1]
```diff
@@ -299,6 +301,17 @@ - (void)setElevation:(CGFloat)elevation {
   _bottomShadow.shadowOpacity = shadowMetrics.bottomShadowOpacity;
 }
 
+- (void)setCornerRadius:(CGFloat)cornerRadius {
+  BOOL radiusChanged = !MDCCGFloatEqual(self.cornerRadius, cornerRadius);
```
Comment (codeman7) [inline components/ShadowLayer/src/MDCShadowLayer.m:305]: Question: In a setter shouldn't we use the ivar?

**Review by codeman7 — APPROVED** [review]

**Review by yarneo — APPROVED** [review]
Let's just make sure we have a tracking of the work needed (if needed) for MDCShapedShadowLayer as well.

File: components/ShadowLayer/src/MDCShadowLayer.m (hunk 1 above)
Comment (romoore) [reply components/ShadowLayer/src/MDCShadowLayer.m:305]: ~Yes we should.~ Ah, no. Only in the getter should we avoid accessing the getter. Sorry!

Comment (romoore) [reply components/ShadowLayer/src/MDCShadowLayer.m:305]: Drive-by comment: Let's please add a unit test for this method. To make sure there isn't an infinite loop. :)

Comment (ianegordon) [reply components/ShadowLayer/src/MDCShadowLayer.m:305]: Yes, typically we prefer to use the ivar in init* methods and getters / setters.
However, in this case, cornerRadius is exposed by CALayer, our super class, so we don't have direct access to the internal data structure storing cornerRadius.

**Comment by ianegordon** [comment]
Note the cornerRadius changes are not getting reflected in the before version.

Before:
shadowradius-broken

After:
shadowradius-fixed

File: components/ShadowLayer/src/MDCShadowLayer.m (hunk 1 above)
Comment (romoore) [reply components/ShadowLayer/src/MDCShadowLayer.m:305]: Maybe next time it can make things clearer to use `super.cornerRadius`?

**Comment by jverkoey** [comment]
@ianegordon You'll be OOO for the next bit. Either land or close this PR please.

**Comment by jverkoey** [comment]
Closing this PR for now. @ianegordon please re-open if you still intend to pursue this.

**Pull request merged**